extension/storage/redisstorageextension/extension.go

```diff
@@ -95,6 +95,25 @@ func (rc redisClient) Batch(ctx context.Context, ops ...*storage.Operation) erro
 		}
 	}
 	_, err := p.Exec(ctx)
+	// once the pipeline has been executed, we need to fetch all the values
+	// and set them on the op
+	for _, op := range ops {
+		switch op.Type {
+		case storage.Get:
+			value, err := rc.client.Get(ctx, rc.prefix+op.Key).Bytes()
+			if errors.Is(err, redis.Nil) {
+				continue
+			}
```

Comment:
```suggestion
			value, err := rc.Get(ctx, op.Key)
			if errors.Is(err, redis.Nil) {
				continue
			}
			if err != nil {
			      return err
			}
```

Couple of questions:
1. Should we handle the `err` here when it's not `redis.Nil`?
2. We can make use of `redisClient.Get(..)` directly I guess?